### PR TITLE
feat: Add goreleaser

### LINKED ISSUE #1834: [infra] Set up a `gnoland` binary release schedule
## Description

This issue concerns creating a CI that will create a release of the `gnoland` binary in some predefined interval, for the purpose of multinode validators pulling it in and running it on their machines.

The CI should create a draft release, and wait for an engineer from the core team to publish it after validation. 
The release should include:
- package deployment to GH Packages
- binary built for different architectures (downloadable archives)
- a change-log detailing what PRs that went into the release

### LINKED ISSUE #2091: CI: Remove custom docker builds and use goreleaser
## Description

Use goreleaser to build docker releases and to standardize the release process.

### PR DESCRIPTION
**Pull request by ajnavarro**
Added goreleaser for main go project only (gno, gnoland, gnoweb, gnokey...). To add other Go projects like faucet we need goreleaser-pro.

Also added the creation of nightly builds that can be used for anyone to test the latest features.

I run it locally using `goreleaser release --snapshot --clean --config .github/goreleaser.yaml`

Tested generated docker images.

Closes #2091 
Closes #1834

<details><summary>Contributors' checklist...</summary>

- [x] Added new tests, or not needed, or not feasible
- [x] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [x] Updated the official documentation or not needed
- [x] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [x] Added references to related issues and PRs
- [x] Provided any useful hints for running manual tests
- [x] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Review by gfanton — APPROVED** [review]
LGTM 🚀

File: .github/goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,481 @@
+project_name: gno
```
Comment (zivkovicmilos) [inline .github/goreleaser.yaml:1]: We should be triggering this workflow only if the actor is a member of @gnolang/tech-staff 

`if github.actor == ...`, I'm not sure what the syntax would be for a member of a GH group, or if that's even possible

**Review by zivkovicmilos — APPROVED** [review]
🚀 🚀 🚀 

cc @mazzy89 @albttx time to quantum jump into the future 🫡

File: .github/goreleaser.yaml (hunk 1 above)
Comment (ajnavarro) [reply .github/goreleaser.yaml:1]: let me check if that is possible

Comment (ajnavarro) [reply .github/goreleaser.yaml:1]: @zivkovicmilos 'github.actor' will contain the username, not the group, so we cannot do what you proposed.

**Pull request merged**